docs/mkdocs/documentation/deploy_kmod.md

```diff
@@ -50,6 +50,48 @@ The reconciliation loop for `Module` runs the following steps:
     2. successful build jobs;
     3. successful signing jobs.
 
+### Soft dependencies between kernel modules
+
+Some setups may require that several kernel modules be loaded to work properly, although the modules do not directly
+depend on each other through symbols.
+`depmod` is usually not aware of those dependencies, and they do not appear in the files it produces.  
+If `mod_a` has a soft dependency on `mod_b`, `modprobe mod_a` will not load `mod_b`.
+
+Soft dependencies can be declared in the `Module` CRD via the
+`.spec.moduleLoader.container.modprobe.modulesLoadingOrder` field:
+
+```yaml
+modulesLoadingOrder:  # optional
+  - mod_a
+  - mod_b
+```
+
+With the configuration above:
+
+- the loading order will be `mod_b`, then `mod_a`;
+- the unloading order will be `mod_a`, then `mod_b`.
+
+### Replacing an in-tree module
+
+Some modules loaded by KMM may replace in-tree modules already loaded on the node.  
+To unload an in-tree module before loading your module, set the `.spec.moduleLoader.container.inTreeModuleToRemove`:
+
+```yaml
+spec:
+  moduleLoader:
+    container:
+      modprobe:
+        moduleName: mod_a
+        # ...
+
+      # Other fields removed for brevity
+      inTreeModuleToRemove: mod_a
+```
+
+The ModuleLoader pod will first try to unload the in-tree `mod_a` before loading the one contained in the ModuleLoader
+image.  
+When the ModuleLoader pod is terminated and its `mod_a` is unloaded, the in-tree `mod_a` will not be loaded again.
```

Comment:
I think we should specify that the module in the InTreeModuleToRemove field can be different from the one in the moduleName